Translate global-css.md (#164)
gatsbyjs/gatsby-pt-BR: docs/docs/global-css.md

---
title: Estilo padrão com arquivos CSS globais
---

Tradicionalmente, os sites são estilizados utilizando arquivos CSS globais.

> As regras de escopo global são declaradas em arquivos `.css` externos, [Especificidade CSS](https://developer.mozilla.org/pt-BR/docs/Web/CSS/Specificity) e o [Cascade](https://developer.mozilla.org/pt-BR/docs/Web/CSS/Cascade) determinam como os estilos são aplicados.

Reviewer: ```suggestion
As regras de escopo global são declaradas em arquivos `.css` externos, e a [Especificicação CSS](https://developer.mozilla.org/pt-BR/docs/Web/CSS/Specificity) e [Cascade](https://developer.mozilla.org/pt-BR/docs/Web/CSS/Cascade) determinam como os estilos são aplicados.
```

## Adicionando estilos globais com um componente de layout

A melhor maneira de adicionar estilos globais é com um [componente de layout compartilhado](/tutorial/part-three/#your-first-layout-component). Esse componente é utilizado para itens que são compartilhados em todo o site, incluindo estilos, componentes do cabeçalho e outros itens comuns.

> > **NOTA:** Esse padrão é implementado em [o padrão inicial](https://github.com/gatsbyjs/gatsby-starter-default/blob/02324e5b04ea0a66d91c7fe7408b46d0a7eac868/src/layouts/index.js#L6).

Reviewer: ```suggestion
> **NOTA:** Esse padrão é implementado [no _starter_ padrão](https://github.com/gatsbyjs/gatsby-starter-default/blob/02324e5b04ea0a66d91c7fe7408b46d0a7eac868/src/layouts/index.js#L6).
```

> Para criar um layout compartilhado com estilos globais, comece criando um novo site Gatsby com o [hello world inicial](https://github.com/gatsbyjs/gatsby-starter-hello-world).

Reviewer: ```suggestion
Para criar um layout compartilhado com estilos globais, comece criando um novo _site_ Gatsby com o [_starter_hello world](https://github.com/gatsbyjs/gatsby-starter-hello-world).
```

}
```

E então, inclua a sua folha de estilos no arquivo `gatsby-browser.js` do seu site.

> **NOTA:** Essa solução somente funciona ao incluir o css, pois esses estilos são extraídos ao criar o JavaScript, mas não para o css-in-js.
> Incluir estilos em um componente de layout ou um global-styles.js é a melhor forma de fazer isso.

```javascript:title=gatsby-browser.js
import "./src/styles/global.css"

// ou:
// require('./src/styles/global.css')
```

> _Nota: Você pode usar o require do Node.js ou o import. Além disso, o local do arquivo css na pasta `src/styles` é de sua escolha._

> Você verá seus estilos globais dando efeito em seu site:

Reviewer: ```suggestion
Você verá que seus estilos globais estão tendo efeito em seu site:
```

import "css/menu.css"
```

> Essa abordagem pode simplificar a integração dos estilos CSS ou [Sass](/packages/gatsby-plugin-sass/) no seu site em Gatsby, permitindo que os membro da equipe consumam e escrevam um CSS mais tradicional baseado em classes. No entando, existem [contrapartidas](#limitations) que devem ser consideradas com relação ao desempenho web e à falta de exclusão em códigos não utilizados.

Reviewer: Adotamos essa convenção de "_site_ Gatsby" no projeto.

```suggestion
Essa abordagem pode simplificar a integração dos estilos CSS ou [Sass](/packages/gatsby-plugin-sass/) no seu _site_ Gatsby, permitindo que os membro da equipe consumam e escrevam um CSS mais tradicional baseado em classes. No entando, existem [contrapartidas](#limitations) que devem ser consideradas com relação ao desempenho web e à falta de exclusão em códigos não utilizados.
```

### Adicionando classes nos componentes

> Como `class` é uma palavra reservada em JavaScript, você irá precisar usar o prop `className`, que será renderizado como um atributo `class` em seu HTML e é suportado pelo seu navegador.

Reviewer: Acho que ficou um pouco melhor. Que tal?

```suggestion
Como `class` é uma palavra reservada em JavaScript, você irá precisar utilizar o prop `className`, que será renderizado como um atributo, suportado pelo navegador, `class` em seu HTML.
```
